### PR TITLE
feat: Form scrollToFirstError support ScrollLogicalPosition

### PR DESCRIPTION
**Pull request by vouis**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/27877

### 💡 Background and solution

control first error field scroll into view 
控制滚动到第一个错误提示的视图位置
for example:
``` js
<Form
  scrollToFirstError={{
    block: 'center'
  }}
>
</Form>
```
{ block:`center` \| `end` \| `nearest`\| `start` }

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Form scrollToFirstError support ScrollLogicalPosition        |
| 🇨🇳 Chinese |    表格组件的scrollToFirstError属性支持 ScrollLogicalPosition 滚动位置参数       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

[components/transfer/index.zh-CN.md](https://github.com/vouis/ant-design/blob/%2327877-feature/components/form/index.zh-CN.md)
[components/transfer/index.en-US.md](https://github.com/vouis/ant-design/blob/%2327877-feature/components/form/index.en-US.md)

-----
[View rendered components/form/index.en-US.md](https://github.com/vouis/ant-design/blob/#27877-feature/components/form/index.en-US.md)
[View rendered components/form/index.zh-CN.md](https://github.com/vouis/ant-design/blob/#27877-feature/components/form/index.zh-CN.md)

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

File: components/form/Form.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export interface FormProps<Values = any> extends Omit<RcFormProps<Values>, 'form
   wrapperCol?: ColProps;
   form?: FormInstance<Values>;
   size?: SizeType;
-  scrollToFirstError?: boolean;
+  scrollToFirstError?: { block?: ScrollLogicalPosition };
```
Comment (hengkx) [inline components/form/Form.tsx:27]: 我看文档里面的 默认值 还是 `false`?

Comment (aLIEzsss4) [reply components/form/Form.tsx:27]: 没有定义ScrollLogicalPosition类型
应当支持所有scrollIntoView. options

File: components/form/index.en-US.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ High performance Form component with data scope management. Including data colle
 | name | Form name. Will be the prefix of Field `id` | string | - |  |
 | preserve | Keep field value even when field removed | boolean | true | 4.4.0 |
 | requiredMark | Required mark style. Can use required mark or optional mark. You can not config to single Form.Item since this is a Form level config | boolean \| `optional` | true | 4.6.0 |
-| scrollToFirstError | Auto scroll to first failed field when submit | boolean | false |  |
+| scrollToFirstError | Auto scroll to first failed field when submit | boolean \| { block:`center` \| `end` \| `nearest`\| `start` } | false |  |
```
Comment (xrkffgg) [inline components/form/index.en-US.md:33]:
```suggestion
| scrollToFirstError | Auto scroll to first failed field when submit | boolean \| { block: `center` \| `end` \| `nearest`\| `start` } | false |  |
```

File: components/form/index.zh-CN.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/ORmcdeaoO/Form.svg
 | name | 表单名称，会作为表单字段 `id` 前缀使用 | string | - |  |
 | preserve | 当字段被删除时保留字段值 | boolean | true | 4.4.0 |
 | requiredMark | 必选样式，可以切换为必选或者可选展示样式。此为 Form 配置，Form.Item 无法单独配置 | boolean \| `optional` | true | 4.6.0 |
-| scrollToFirstError | 提交失败自动滚动到第一个错误字段 | boolean | false |  |
+| scrollToFirstError | 提交失败自动滚动到第一个错误字段 | boolean \| { block:`center` \| `end` \| `nearest`\| `start` } | false |  |
```
Comment (xrkffgg) [inline components/form/index.zh-CN.md:34]:
```suggestion
| scrollToFirstError | 提交失败自动滚动到第一个错误字段 | boolean \| { block: `center` \| `end` \| `nearest`\| `start` } | false |  |
```

**Pull request merged**